ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -324,25 +326,47 @@ public static String getSqlTypeName(final Schema schema) {
   private static String getStructString(final Schema schema) {
     return schema.fields().stream()
         .map(field -> field.name() + " " + getSqlTypeName(field.schema()))
-        .collect(Collectors.joining(", ", "STRUCT <", ">"));
+        .collect(Collectors.joining(", ", "STRUCT<", ">"));
   }
 
   static org.apache.avro.Schema buildAvroSchema(final Schema schema, final String name) {
+    return buildAvroSchema(DEFAULT_NAMESPACE, name, schema);
+  }
+
+  private static org.apache.avro.Schema buildAvroSchema(
+      final String namespace,
+      final String name,
+      final Schema schema
+  ) {
+    final String avroName = avroify(name);
     final FieldAssembler<org.apache.avro.Schema> fieldAssembler = org.apache.avro.SchemaBuilder
-        .record(name).namespace("ksql")
+        .record(avroName).namespace(namespace)
         .fields();
 
     for (final Field field : schema.fields()) {
+      final String fieldName = avroify(field.name());
+      final String fieldNamespace = namespace + "." + avroName;
+
       fieldAssembler
-          .name(field.name().replace(".", "_"))
-          .type(getAvroSchemaForField(field.schema()))
+          .name(fieldName)
+          .type(getAvroSchemaForField(fieldNamespace, fieldName, field.schema()))
           .withDefault(null);
     }
 
     return fieldAssembler.endRecord();
   }
 
-  private static org.apache.avro.Schema getAvroSchemaForField(final Schema fieldSchema) {
+  private static String avroify(final String name) {
```

Comment:
This method extends the existing replacement of `.` characters in field names, (which Avro doesn't like), to also include `-` characters, as otherwise our own tests fail.  There are probably other characters we should replace, but that's for another PR.